src/MassTransit/Transports/ReceiveTransport.cs

```diff
@@ -153,7 +153,10 @@ async Task Run()
                         }
 
                         if (IsStopping)
+                        {
+                            await _context.OnTransportStop(Stopped).ConfigureAwait(false);
```

Comment:
Feels like there has to be some cancellation other than Stopped, since that's only set when this loop completes.